platform/javascript/js/engine/preloader.js

```diff
@@ -79,51 +77,47 @@ var Preloader = /** @constructor */ function() { // eslint-disable-line no-unuse
 		if (loaded !== lastProgress.loaded || total !== lastProgress.total) {
 			lastProgress.loaded = loaded;
 			lastProgress.total = total;
-			if (typeof progressFunc === 'function')
-				progressFunc(loaded, total);
+			if (typeof progressFunc === 'function') progressFunc(loaded, total);
 		}
-		if (!progressIsFinal)
-			requestAnimationFrame(animateProgress);
-	}
+		if (!progressIsFinal) requestAnimationFrame(animateProgress);
+	};
 
 	this.animateProgress = animateProgress;
 
-	this.setProgressFunc = function(callback) {
+	this.setProgressFunc = function (callback) {
 		progressFunc = callback;
-	}
-
+	};
 
-	this.loadPromise = function(file) {
-		return new Promise(function(resolve, reject) {
+	this.loadPromise = function (file) {
+		return new Promise(function (resolve, reject) {
 			loadXHR(resolve, reject, file, loadingFiles, DOWNLOAD_ATTEMPTS_MAX);
 		});
-	}
+	};
 
 	this.preloadedFiles = [];
-	this.preload = function(pathOrBuffer, destPath) {
+	this.preload = function (pathOrBuffer, destPath) {
 		let buffer = null;
 		if (typeof pathOrBuffer === 'string') {
-			var me = this;
-			return this.loadPromise(pathOrBuffer).then(function(xhr) {
+			const me = this;
+			return this.loadPromise(pathOrBuffer).then(function (xhr) {
 				me.preloadedFiles.push({
 					path: destPath || pathOrBuffer,
-					buffer: xhr.response
+					buffer: xhr.response,
 				});
 				return Promise.resolve();
 			});
-		} else if (pathOrBuffer instanceof ArrayBuffer) {
+		} if (pathOrBuffer instanceof ArrayBuffer) {
```

Comment:
Another `else if` transformed to `if`. Here I guess it's detecting that both conditions can't be true at the same time, so the `else` is not necessary. The `if` should likely be on a newline though, unless that's a typical JS formatting for "would be `else if` but linter is clever so it's `if`" situations :P

Reply:
I think I'll just disable that rule for `else if`s. It makes a mess, I prefer being explicit.
I'll see if leaving the one for the final `else` works okay.